bot/seasons/evergreen/battleship.py

```diff
@@ -107,211 +111,227 @@ def __init__(
+        return "\n".join([LETTERS] + rows)
+
+    @staticmethod
+    def get_square(grid: Grid, square: str) -> Square:
         """Grabs a square from a grid with an inputted key."""
-        index = {"A": 0, "B": 1, "C": 2, "D": 3, "E": 4, "F": 5, "G": 6, "H": 7, "I": 8, "J": 9}
-        letter = square[0]
+        index = ord(square[0]) - ord("A")
         number = int(square[1:])
 
-        return grid[number-1][index[letter]]  # -1 since lists are indexed from 0
+        return grid[number-1][index]  # -1 since lists are indexed from 0
 
     async def game_over(
         self,
         *,
-        timeout: bool = False,
-        winner: typing.Optional[discord.Member] = None,
-        loser: typing.Optional[discord.Member] = None
+        winner: discord.Member,
+        loser: discord.Member
     ) -> None:
         """Removes games from list of current games and announces to public chat."""
-        if not timeout:  # If someone won and not the game timed out
-            await self.public_channel.send(f"Game Over! {winner.mention} won against {loser.mention}")
-            self_grid_1 = self.format_grid([
-                [ship_emojis[bool(square.boat), square.aimed] for square in row]
-                for row in self.grids[self.player1]
-            ])
-
-            self_grid_2 = self.format_grid([
-                [ship_emojis[bool(square.boat), square.aimed] for square in row]
-                for row in self.grids[self.player2]
-            ])
-
-            await self.public_channel.send(f"{self.player1}'s Board:\n{self_grid_1}")
-            await self.public_channel.send(f"{self.player2}'s Board:\n{self_grid_2}")
+        await self.public_channel.send(f"Game Over! {winner.mention} won against {loser.mention}")
 
-        self.bot.get_cog("Battleship").games.remove(self)
+        for player in (self.p1, self.p2):
+            grid = self.format_grid(player, SHIP_EMOJIS)
+            await self.public_channel.send(f"{player.user}'s Board:\n{grid}")
 
     @staticmethod
-    def check_sink(grid: grid_typehint, boat: str) -> bool:
+    def check_sink(grid: Grid, boat: str) -> bool:
         """Checks if all squares containing a given boat have sunk."""
         return all(square.aimed for row in grid for square in row if square.boat == boat)
 
     @staticmethod
-    def check_gameover(grid: grid_typehint) -> bool:
+    def check_gameover(grid: Grid) -> bool:
         """Checks if all boats have been sunk."""
         return all(square.aimed for row in grid for square in row if square.boat)
 
     def setup_grids(self) -> None:
         """Places the boats on the grids to initialise the game."""
-        for player in (self.player1, self.player2):
-            for name, size in ships.items():
+        for player in (self.p1, self.p2):
+            for name, size in SHIPS.items():
                 while True:  # Repeats if about to overwrite another boat
                     overwrite = False
```

Comment:
This isn't a good name. It's not clear what it's overwriting (the `coords` list I guess). It's more of a "keep trying new coords cause the previous collided with an existing ship" so perhaps `ship_collision` is a better name?